diff --git a/source3/wscript b/source3/wscript
index a2164cc..5353b91 100644
--- a/source3/wscript
+++ b/source3/wscript
@@ -940,7 +940,7 @@ syscall(SYS_setgroups32, 0, NULL);
                                 ''',
                                 'HAVE_LINUX_THREAD_CREDENTIALS',
                                 addmain=False,
-                                execute=True,
+                                execute=False,
                                 msg="Checking whether we can use Linux thread-specific credentials with 32-bit system calls")
         else:
             seteuid = conf.CHECK_CODE('''
@@ -951,7 +951,7 @@ syscall(SYS_setgroups32, 0, NULL);
                                 ''',
                                 'HAVE_LINUX_THREAD_CREDENTIALS',
                                 addmain=False,
-                                execute=True,
+                                execute=False,
                                 msg="Checking whether we can use Linux thread-specific credentials")
     if not seteuid:
         seteuid = conf.CHECK_CODE('''
